kube/src/api/typed.rs

```diff
@@ -293,6 +293,10 @@ where
     /// This returns a future that awaits the initial response,
     /// then you can stream the remaining buffered `WatchEvent` objects.
     ///
+    /// Note that a watch call __will terminate in <5 minutes__, based on [`ListParams::timeout`],
+    /// and as such, is best suited for __short term__ watches.
+    /// For continual tracking of resources, consider a managed [`watcher`].
```

Comment:
1. The timeout parameter is just a hint to the server, which can really enforce whatever timeout it wants, so (in the long term) there isn't really a reliable threshold here for when `Api::watch` is safe
2. There are other causes for errors here (network interruption, apiserver bugs, network topology changes, restarts), that you'd probably also want to recover from

Reply:
Yeah, that is true, probably need to go through #334 and check how this interacts without defaults now. We should possibly remove the hard limit.

You are right. I'll reword it a bit. Possibly we should have some more convenience (or at least examples) around `watcher`s that allow the same type of watch with an explicit timeout (without having to consider the other error cases).